pivot.py

```diff
@@ -0,0 +1,1723 @@
+        ]
+
+    def render(self):
+        pass
+
+    def header_x(self):
+        pool = Pool()
+        PivotHeader = pool.get('www.pivot_header')
+
+        fields = []
+        if self.table_properties != 'null':
+            for table_property in self.table_properties.split('__'):
+                if table_property:
+                    tp = {}
+                    for field in table_property.split('&'):
+                        if field:
+                            tp[field.split('=')[0]] = field.split('=')[1]
+                    if tp['position'] == 'x':
+                        fields.append(tp)
+
+            if fields:
+                fields = sorted(fields, key=lambda x: x['hierarchy'])
+
+        self.header = 'x'
+        with div(cls="px-4 sm:px-6 lg:px-8 mt-8 flow-root col-span-4", id='header_x') as header_x:
+            div(id='field_selection_x')
+            with div(cls="-mx-4 -my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"):
+                with div(cls="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"):
+                    with table(cls="min-w-full divide-y divide-gray-300"):
+                        with thead():
+                            with tr():
+                                th(scope="col", cls="relative py-3.5 pl-3 pr-4 sm:pr-0").add(
+                                    raw('<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="size-6" transform="rotate(90)"><path stroke-linecap="round" stroke-linejoin="round" d="M9 4.5v15m6-15v15m-10.875 0h15.75c.621 0 1.125-.504 1.125-1.125V5.625c0-.621-.504-1.125-1.125-1.125H4.125C3.504 4.5 3 5.004 3 5.625v12.75c0 .621.504 1.125 1.125 1.125Z" /></svg>'))
+                                th('X axis', scope="col", cls="py-3.5 pl-4 pr-3 text-left text-sm font-semibold text-gray-900 sm:pl-0")
+                                th(scope="col", cls="relative py-3.5 pl-3 pr-4 sm:pr-0").add(span('Up', cls="sr-only"))
+                                th(scope="col", cls="relative py-3.5 pl-3 pr-4 sm:pr-0").add(span('Down', cls="sr-only"))
+                                with th(scope="col", cls="relative py-3.5 pl-3 pr-4 sm:pr-0"):
+                                    a(href="#", cls="text-indigo-600 hover:text-indigo-900",
+                                        hx_target="#field_selection_x",
+                                        hx_post=PivotHeader(header='x', database_name=self.database_name, table_name=self.table_name, table_properties=self.table_properties, render=False).url('open_field_selection'),
+                                        hx_trigger="click", hx_swap="outerHTML").add(raw('<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="size-6"><path stroke-linecap="round" stroke-linejoin="round" d="M12 4.5v15m7.5-7.5h-15" /></svg>'))
```

Comment:
Better use constants for each SVG. You can even include `raw()` in the assignment.